src/compiler.ts

```diff
@@ -8700,27 +8714,122 @@ export class Compiler extends DiagnosticEmitter {
       }
       let fieldInstance = <Field>member;
       let fieldType = fieldInstance.type;
-      exprs[i + 1] = this.module.store( // TODO: handle setters as well
-        fieldType.byteSize,
-        this.module.local_get(tempLocal.index, this.options.nativeSizeType),
-        this.compileExpression(values[i], fieldInstance.type, Constraints.CONV_IMPLICIT),
-        fieldType.toNativeType(),
-        fieldInstance.memoryOffset
-      );
+      exprs.push(this.module.store( // TODO: handle setters as well
+       fieldType.byteSize,
+       this.module.local_get(tempLocal.index, this.options.nativeSizeType),
+       this.compileExpression(values[i], fieldInstance.type, Constraints.CONV_IMPLICIT),
+       fieldType.toNativeType(),
+       fieldInstance.memoryOffset
+      ));
```

Comment:
This looks problematic, but already was so before so nothing introduced by this PR. If `fieldType` is managed here, and the class is managed as well, we need to retain a reference that the instance now holds, ideally accounting for pre-retained values (from function calls or ctors) which we can trigger by `WILL_RETAIN` and checking `skippedAutoreleases`.

```suggestion
      let expr = this.compileExpression(values[i], fieldType, Constraints.CONV_IMPLICIT | Constraints.WILL_RETAIN);
      if (isManaged && fieldType.isManaged && !this.skippedAutoreleases.has(expr)) {
        expr = this.makeRetain(expr);
      }
      exprs.push(this.module.store( // TODO: handle setters as well
        fieldType.byteSize,
        this.module.local_get(tempLocal.index, this.options.nativeSizeType),
        expr,
        fieldType.toNativeType(),
        fieldInstance.memoryOffset
      ));
```